### PR TITLE
Volume Command

### PR DESCRIPTION
**Pull request by Ikinon**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Feature
Music - Volume Setting

### CODE REVIEW

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        CascadePlayer player = context.getData().getMusicPlayer();
+        if (context.getArgs().length == 0) {
+            context.getTypedMessaging().replyInfo("Current volume is %s%%", player.getPlayer().getVolume());
+            return;
+        }
+        Integer Arguments = Integer.valueOf(context.getMessage(0));
+        if (Arguments > 100 || Arguments < 0) {
+            context.getTypedMessaging().replyWarning("Volume needs to be between 100 and 0");
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: flip this around. aka make it `0 and 100` not `100 and 0`

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+public class VolumeCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        CascadePlayer player = context.getData().getMusicPlayer();
+        if (context.getArgs().length == 0) {
+            context.getTypedMessaging().replyInfo("Current volume is %s%%", player.getPlayer().getVolume());
+            return;
+        }
+        Integer Arguments = Integer.valueOf(context.getMessage(0));
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: 1. why is this name Argument`s`
2. camelCase

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public String command() {
+        return "volume";
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return CascadePermission.of("Volume", "search", getModule());
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Thats not how this works. the second argument is the perm so switch that to volume as well

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java (hunk 2 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: oh also you should parse this better.
first off make the variable an int second off parse it with Integer.parseInt() and surround that in a try catch

**Comment by binaryoverload** [comment]
You need to use the method that specifys the flag to restrict this to Patreon guilds

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java (hunk 2 above)
Comment (Ikinon) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: `why is this name Arguments` I don't know what else to name it to. Suggestion?

Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Why is getMessage being used? We have a get arg as int method

Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: also you can name it `volume` :P

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+public class VolumeCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        CascadePlayer player = context.getData().getMusicPlayer();
+        if (context.getArgs().length == 0) {
+            context.getTypedMessaging().replyInfo("Current volume is %s%%", player.getPlayer().getVolume());
+            return;
+        }
+        Integer Volume = null;
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: 1. int not Integer
2. camelCase

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 5]
```diff
@@ -16,23 +16,21 @@ public void onCommand(Member sender, CommandContext context) {
...
-
-        } catch (java.lang.NumberFormatException e) {
+        int volume = -1;
+        if(context.isArgInteger(0)) {
+            volume = context.getArgAsInteger(0);
             context.getUIMessaging().replyUsage(this);
             return;
         }
-        if (Volume < 0 || Volume > 100) {
+        if (volume < 0 || volume > 100) {
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Why don't we just let them go 150 or smth?

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 6]
```diff
@@ -16,23 +16,21 @@ public void onCommand(Member sender, CommandContext context) {
...
             context.getTypedMessaging().replyWarning("Volume needs to be between 100 and 0");
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Swap these two round i.e `between 0 and 100`

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 7]
```diff
@@ -16,23 +16,21 @@ public void onCommand(Member sender, CommandContext context) {
...
             return;
         }
 
-        if (Volume == context.getData().getMusicPlayer().getPlayer().getVolume()) {
+        if (volume == context.getData().getMusicPlayer().getPlayer().getVolume()) {
             context.getTypedMessaging().replyInfo("Volume is already %s%%", player.getPlayer().getVolume());
         } else {
-            player.getPlayer().setVolume(Volume);
+            player.getPlayer().setVolume(volume);
             context.getTypedMessaging().replyInfo("Volume set to %s%%", player.getPlayer().getVolume());
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Use `%d` this is a number not a string

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java [hunk 8]
```diff
@@ -16,23 +16,21 @@ public void onCommand(Member sender, CommandContext context) {
...
         }
-        if (Volume < 0 || Volume > 100) {
+        if (volume < 0 || volume > 100) {
             context.getTypedMessaging().replyWarning("Volume needs to be between 100 and 0");
             return;
         }
 
-        if (Volume == context.getData().getMusicPlayer().getPlayer().getVolume()) {
+        if (volume == context.getData().getMusicPlayer().getPlayer().getVolume()) {
             context.getTypedMessaging().replyInfo("Volume is already %s%%", player.getPlayer().getVolume());
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Same as below

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java (hunk 6 above)
Comment (Ikinon) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Umm, is it not that already?

Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: in the text not the if statment

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java (hunk 5 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: I think we can't. let me double check though.

File: src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java (hunk 6 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: In fact switch the if statement back around because now thats broken

Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: The if statement seems fine to me?

Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/music/VolumeCommand.java]: Whops nvm

**Pull request merged**